account_asset_management/models/account_asset.py

```diff
@@ -526,9 +526,22 @@ def group_lines(x, y):
             # recompute in case of deviation
             depreciated_value_posted = depreciated_value = 0.0
             if posted_lines:
+                total_table_lines = sum(
+                    [len(entry[1]["lines"]) for entry in enumerate(table)]
+                )
+                move_check_lines = asset.depreciation_line_ids.filtered(
+                    lambda x: x.move_check
+                )
                 last_depreciation_date = last_line.line_date
                 last_date_in_table = table[-1]['lines'][-1]['date']
-                if last_date_in_table <= last_depreciation_date:
+                # If the number of lines in the table is the same as the depreciation
+                # lines, we will not show an error even if the dates are the same.
+                if (
```

Comment:
I don't get why is this needed, as this is checking the integrity. We should keep it or provide a case where it's legit to bypass it for me to analyze it.

Reply:
I think an example will help explain the need for this change:

**Example A**:
Test asset with a value of 10.
Depreciation table generated:
- 01/01/2021 - 5
- 02/01/2021 - 5

If we regenerate (Compute button) the depreciation table, there is no problem. If we generate the entries for all depreciations, the asset is closed.

**Example B**:
Test asset with a value of 10.
Depreciation table generated:
- 01/01/2021 - 5
- 02/01/2021 - 5

We modify the value of last depreciation to 4.95

If we regenerate the depreciation table, there is no problem. If we generate the entries for all amortizations, the asset is NOT closed because 0.05 will remain pending.

We generate the depreciation table again and the error would already be obtained (the saved date of the last depreciation is equal to the date of the last depreciation that Odoo calculates).